app/src/main/java/csustan/edu/cs4800/warriordiningapp/MainActivity.java

```diff
@@ -1,13 +1,81 @@
 package csustan.edu.cs4800.warriordiningapp;
 
 import androidx.appcompat.app.AppCompatActivity;
+import androidx.core.app.NotificationCompat;
+import androidx.core.app.NotificationManagerCompat;
 
+import android.app.NotificationChannel;
+import android.app.NotificationManager;
+import android.os.Build;
 import android.os.Bundle;
 import android.view.View;
 import android.widget.Button;
 
 public class MainActivity extends AppCompatActivity implements View.OnClickListener {
 
+    Button notificationButton;
+    public boolean isCancel = true;
+    public boolean activeRequest = false; //use boolean active request for notification
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        notificationButton = findViewById(R.id.notificationButton); //references notificationButton in xml by id
+
+        //special devices compatibility if statement
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            NotificationChannel channel = new NotificationChannel("Notification1", "Itemnotification", NotificationManager.IMPORTANCE_DEFAULT);
+            NotificationManager manager = getSystemService(NotificationManager.class);
+            manager.createNotificationChannel(channel);
+        }
+
+        notificationButton.setOnClickListener(new View.OnClickListener() {
+                                                  @Override
+                                                  public void onClick(View v) {
+                                                      //notification code
```

Comment:
Nice demo code to display a local notification. In the future, this button will need to do a couple things, so they should probably be encapsulated in a controller class (layered & MVC architecture patterns):
1. If there's an existing notification request for this item, delete it (or mark it canceled...we should discuss further) and stop processing.
2. Otherwise, If the user hasn't already given notification permissions for our app, request them from the operating system to get a notification token.
3. Post to the server to create a notification request using the token obtained in step 2.